src/provisioner/config_store.py

```diff
@@ -30,9 +30,16 @@ def set_kvs(self, kvs: list):
         Parameters:
         kvs - List of KV tuple, e.g. [('k1','v1'),('k2','v2')]
         """
-
-        for key, val in kvs:
-            Conf.set(self._conf_idx, key, val)
+        for key, val in kvs.copy():
+            try:
+                Conf.set(self._conf_idx, key, val)
+                kvs.remove((key, val))
+            except AssertionError:
+                # For consul(confstore backend),
+                # value should be string type.
+                key_value_list = self._parse_key_values(kvs)
+                for key, val in key_value_list:
```

Comment:
This is not enough. We need to avoid treating it as hierarchical config. 
kvs means set of key value in strings. 

If we assume non string then there would be another exception in second conf.set()

Reply:
Removed this code, as it is not required anymore.